src/install_wheel.rs

```diff
@@ -0,0 +1,194 @@
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub(crate) enum LibKind {
+    /// Install into the `purelib` directory.
+    Pure,
+    /// Install into the `platlib` directory.
+    Plat,
+    /// Unknown wheel kind when major version
+    /// for `Wheel-Version: 1.0`
+    /// is greater than 1
+    Unknown,
+}
+
+/// Parse a file with `Key: value` entries such as WHEEL and METADATA
+fn parse_key_value_file(
+    file: impl Read,
+    debug_filename: &str,
+) -> miette::Result<HashMap<String, Vec<String>>> {
+    let mut data: HashMap<String, Vec<String>> = HashMap::default();
+
+    let file = BufReader::new(file);
+    for (line_no, line) in file.lines().enumerate() {
+        let line = line.into_diagnostic()?.trim().to_string();
+        if line.is_empty() {
+            continue;
+        }
+        let (key, value) = line.split_once(':').ok_or_else(|| {
+            miette::miette!(
+                "Line {} of the {debug_filename} file is invalid",
+                line_no + 1
+            )
+        })?;
+        data.entry(key.trim().to_string())
+            .or_default()
+            .push(value.trim().to_string());
+    }
+    Ok(data)
+}
+
+/// Parse WHEEL file.
+///
+/// > {distribution}-{version}.dist-info/WHEEL is metadata about the archive itself in the same
+/// > basic key: value format:
+pub(crate) fn parse_wheel_file(wheel_text: &str) -> miette::Result<LibKind> {
+    // {distribution}-{version}.dist-info/WHEEL is metadata about the archive itself in the same basic key: value format:
+    let data = parse_key_value_file(&mut wheel_text.as_bytes(), "WHEEL")?;
+
+    // Determine whether Root-Is-Purelib == ‘true’.
+    // If it is, the wheel is pure, and should be installed into purelib.
+    let root_is_purelib = data
+        .get("Root-Is-Purelib")
+        .and_then(|root_is_purelib| root_is_purelib.first())
+        .is_some_and(|root_is_purelib| root_is_purelib == "true");
+    let lib_kind = if root_is_purelib {
+        LibKind::Pure
+    } else {
+        LibKind::Plat
+    };
+
+    // mkl_fft-1.3.6-58-cp310-cp310-manylinux2014_x86_64.whl has multiple Wheel-Version entries, we have to ignore that
+    // like pip
+    let wheel_version = data
+        .get("Wheel-Version")
+        .and_then(|wheel_versions| wheel_versions.first());
+    let wheel_version = wheel_version
+        .and_then(|wheel_version| wheel_version.split_once('.'))
+        .ok_or_else(|| miette::miette!("Invalid Wheel-Version in WHEEL file: {wheel_version:?}"))?;
+
+    // pip has some test wheels that use that ancient version,
+    // and technically we only need to check that the version is not higher
+    if wheel_version == ("0", "1") {
+        return Ok(lib_kind);
+    }
+    // Check that installer is compatible with Wheel-Version. Warn if minor version is greater, abort if major version is greater.
+    // Wheel-Version: 1.0
+    if wheel_version.0 != "1" {
+        return Ok(LibKind::Unknown);
+    }
+    Ok(lib_kind)
+}
```

Comment:
Are there any tests we could vendor as well?

Reply:
added necessary vendored tests